coherence-distributed-configuration/src/main/java/io/micronaut/coherence/discovery/CoherenceConfigurationClient.java

```diff
@@ -0,0 +1,182 @@
+
+    public CoherenceConfigurationClient(ApplicationConfiguration applicationConfiguration,
+                                        CoherenceClientConfiguration coherenceClientConfiguration) {
+        this.applicationConfiguration = applicationConfiguration;
+        this.coherenceClientConfiguration = coherenceClientConfiguration;
+        LOG.debug("Coherence Client configuration: " + coherenceClientConfiguration);
+    }
+
+    @Override
+    public Publisher<PropertySource> getPropertySources(Environment environment) {
+        ValueExtractor<String, String> keyExtractor = new KeyExtractor<>();
+        NamedCache<String, Object> configCache = getCache(CONFIG_CACHE_NAME, coherenceClientConfiguration);
+
+        Map<Integer, String> keys = buildKeys(applicationConfiguration, environment);
+        for (Map.Entry<Integer, String> entry : keys.entrySet()) {
+            final Integer priority = entry.getKey();
+            final String propertySource = entry.getValue();
+            final String propertySourcePath = propertySource + "/";
+            Remote.Predicate<String> pathPredicate = cacheKey -> cacheKey.startsWith(propertySourcePath) &&
+                    cacheKey.length() > propertySourcePath.length() &&
+                    !cacheKey.substring(propertySourcePath.length()).contains("/");
+            Filter<String> filter = Filters.predicate(keyExtractor, pathPredicate);
+            Map<String, Object> props = configCache.entrySet(filter)
+                    .stream()
+                    .collect(Collectors.toMap((Converter<Map.Entry<String, Object>, String>)
+                                    cacheEntry -> cacheEntry.getKey().substring(propertySourcePath.length()),
+                            Map.Entry::getValue));
+            Flowable<PropertySource> propertySourceFlowable = Flowable.just(PropertySource.of(propertySource, props, priority));
+            propertySources.add(propertySourceFlowable);
+        }
+        return Flowable.merge(propertySources);
+    }
+
+    /**
+     * Returns NamedCache instance.
+     *
+     * @param cacheName                    cache name
+     * @param coherenceClientConfiguration configuration
+     * @return cache
+     */
+    protected NamedCache<String, Object> getCache(String cacheName, CoherenceClientConfiguration coherenceClientConfiguration) {
+        Session session = buildSession(coherenceClientConfiguration);
+        return session.getCache(cacheName);
+    }
+
+    /**
+     * Builds Coherence session.
+     *
+     * @param coherenceClientConfiguration configuration
+     * @return Coherence session
+     */
+    protected Session buildSession(CoherenceClientConfiguration coherenceClientConfiguration) {
+        Channel channel = buildChannel(coherenceClientConfiguration);
+
+        GrpcSessionConfiguration.Builder builder = GrpcSessionConfiguration.builder(channel);
+        GrpcSessionConfiguration grpcSessionConfiguration = builder.build();
```

Comment:
I believe this is where you need to set the scope to `$SYS`. There may also be a method to `getSystemSession`, in which case that's what you should use so we don't leak the internals.